pyls/uris.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2017 Palantir Technologies, Inc.
+"""A collection of URI utilities with logic built on the VSCode URI library.
+
+https://github.com/Microsoft/vscode-uri/blob/e59cab84f5df6265aed18ae5f43552d3eef13bb9/lib/index.ts
+"""
+import re
+from urllib import parse
+from pyls import IS_WIN
+
+RE_DRIVE_LETTER_PATH = re.compile(r'^\/[a-zA-z]:')
```

Comment:
Ah that's interesting, looks like it's the same in the VSCode package: https://github.com/Microsoft/vscode-uri/blob/e59cab84f5df6265aed18ae5f43552d3eef13bb9/lib/index.ts#L56